src/dcc/DccDebug.cxxtest

```diff
@@ -0,0 +1,157 @@
+}
+
+TEST(DccDebug, Speed28_0)
+{
+    Packet pkt;
+    pkt.set_dcc_speed28(
+        DccShortAddress(42), true, 0);
+    EXPECT_EQ(
+        "[dcc] Short Address 42 SPD F 0", packet_to_string(pkt));
+    pkt.clear();
+    pkt.set_dcc_speed28(
+        DccShortAddress(42), false, 0);
+    EXPECT_EQ(
+        "[dcc] Short Address 42 SPD R 0", packet_to_string(pkt));
+    pkt.clear();
+    pkt.set_dcc_speed28(
+        DccLongAddress(42), false, 0);
+    EXPECT_EQ(
+        "[dcc] Long Address 42 SPD R 0", packet_to_string(pkt));
+}
+
+TEST(DccDebug, Speed28_1)
+{
+    Packet pkt;
+    pkt.set_dcc_speed28(
+        DccShortAddress(42), true, 1);
+    EXPECT_EQ(
+        "[dcc] Short Address 42 SPD F 1", packet_to_string(pkt));
+    pkt.clear();
+    pkt.set_dcc_speed28(
+        DccShortAddress(42), false, 1);
+    EXPECT_EQ(
+        "[dcc] Short Address 42 SPD R 1", packet_to_string(pkt));
+    pkt.clear();
+    pkt.set_dcc_speed28(
+        DccLongAddress(42), false, 1);
+    EXPECT_EQ(
+        "[dcc] Long Address 42 SPD R 1", packet_to_string(pkt));
+}
+
+TEST(DccDebug, Speed28_others)
+{
+    Packet pkt;
+    pkt.set_dcc_speed28(
+        DccShortAddress(42), true, 13);
+    EXPECT_EQ(
+        "[dcc] Short Address 42 SPD F 13", packet_to_string(pkt));
+    pkt.clear();
+    pkt.set_dcc_speed28(
+        DccShortAddress(42), false, 28);
+    EXPECT_EQ(
+        "[dcc] Short Address 42 SPD R 28", packet_to_string(pkt));
+    pkt.clear();
+    pkt.set_dcc_speed28(
+        DccLongAddress(42), false, 28);
+    EXPECT_EQ(
+        "[dcc] Long Address 42 SPD R 28", packet_to_string(pkt));
+}
+
+TEST(DccDebug, F0_4)
+{
+    Packet pkt;
+    pkt.add_dcc_address(DccShortAddress(3));
+    pkt.add_dcc_function0_4(0b10111);
+    EXPECT_EQ(
+        "[dcc] Short Address 3 F[0-4]=11101", packet_to_string(pkt));
+    pkt.clear();
+    pkt.add_dcc_address(DccShortAddress(3));
+    pkt.add_dcc_function0_4(0b10110);
+    EXPECT_EQ(
+        "[dcc] Short Address 3 F[0-4]=01101", packet_to_string(pkt));
+}
+
+TEST(DccDebug, F5_8)
+{
+    Packet pkt;
+    pkt.add_dcc_address(DccShortAddress(3));
+    pkt.add_dcc_function5_8(0b1011);
+    EXPECT_EQ(
+        "[dcc] Short Address 3 F[5-8]=1101", packet_to_string(pkt));
+}
+
+TEST(DccDebug, F9_12)
+{
+    Packet pkt;
+    pkt.add_dcc_address(DccShortAddress(3));
+    pkt.add_dcc_function9_12(0b1101);
+    EXPECT_EQ(
+        "[dcc] Short Address 3 F[9-12]=1011", packet_to_string(pkt));
+}
+
+TEST(DccDebug, F13_20)
+{
+    Packet pkt;
+    pkt.add_dcc_address(DccShortAddress(3));
+    pkt.add_dcc_function13_20(0b10010110);
+    EXPECT_EQ(
+        "[dcc] Short Address 3 F[13-20]=01101001", packet_to_string(pkt));
+}
+
+TEST(DccDebug, F21_28)
+{
+    Packet pkt;
+    pkt.add_dcc_address(DccShortAddress(3));
+    pkt.add_dcc_function21_28(0b10010110);
+    EXPECT_EQ(
+        "[dcc] Short Address 3 F[21-28]=01101001", packet_to_string(pkt));
+}
+
+
+} // namespace
+} // namespace dcc
```

Comment:
I don't see any tests for non-DCC protocol.

Reply:
I have not written the decoding code yet, that's why there are no tests.

Reply:
i did add a dummy test though so we can know we recognize them.